compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -159,6 +135,8 @@ trait ResolverAstLoweringExt {
     fn get_lifetime_res(&self, id: NodeId) -> Option<LifetimeRes>;
     fn take_extra_lifetime_params(&mut self, id: NodeId) -> Vec<(Ident, NodeId, LifetimeRes)>;
     fn decl_macro_kind(&self, def_id: LocalDefId) -> MacroKind;
+    fn record_def_id_remap(&mut self, from: LocalDefId, to: LocalDefId);
```

Comment:
we need comments on these methods -- what do they do? what does remap mean? why would we want to remap?

Probably makes sense to have a larger comment on the `generics_def_id_map` field and then smaller comments on the methods that refer to that one.